src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,229 @@
+    throws SQLException, IOException, InterruptedException, ParseException {
+    String Id = "a03Dn000008bkucIAA";
+    //big query table="myTable7";
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject json = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(json);
+    }
+    SalesforceClient.queryObject(Id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse);
+
+  }
+  public static boolean validateBQToSalesforceRecordValues(String targetTable,String objectName)
+    throws SQLException, IOException, InterruptedException, ParseException {
+    String Id = "a03Dn000008c4D8IAI";
+    //big query table="E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575";
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject json = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(json);
+    }
+    SalesforceClient.queryObject(Id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse);
+
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows)
+    throws IOException, InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData,
+                                                     List<JsonObject> bigQueryData)
+    throws SQLException, ParseException {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
+    int columnCountTarget = 0;
+    if (salesforceData.size() > 0) {
+      columnCountTarget = salesforceData.get(0).entrySet().size();
+    }
+    // Get the column count of the first JsonObject in bigQueryData
+    int jsonObjectIdx = 0;
+    int columnCountSource = 0;
+    if (bigQueryData.size() > 0) {
+      columnCountSource = bigQueryData.get(jsonObjectIdx).entrySet().size();
+    }
+    // Compare the number of columns in the source and target
+//    Assert.assertEquals("Number of columns in source and target are not equal",
+//                        columnCountSource, columnCountTarget);
+
+    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String tableName = "E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575";
```

Comment:
Don't hardcode the table Name value here.

Reply:
done